src/environmentd/src/http.rs

```diff
@@ -1022,6 +1022,31 @@ async fn auth(
                 include_www_authenticate_header,
             });
         }
+        Authenticator::Oidc(oidc) => match creds {
```

Comment:
very preliminary comment: but I think with the number of authenticators growing, we should maybe put this behind an interface, maybe make Authenticator compatible with trait Objects, but at least we could move this big match statement to the authenticator crate.